### PR TITLE
feat(accel_brake_map_calibrator): add new Recursive Least Square method

### PR DESCRIPTION
**Pull request by td12321**
## Description

For accel_map_calibrator, implemented a new method to execute Recursive Least Squares on 4 points around new data.

## Related links

[Gabrielle Lochrie, Michael Doljevic, Mario Nona, Yongsoon Yoon, Anti-Windup Recursive Least Squares Method for Adaptive Lookup Tables with Application to Automotive Powertrain Control Systems, IFAC-PapersOnLine, Volume 54, Issue 20, 2021, Pages 840-845](https://www.sciencedirect.com/science/article/pii/S240589632102320X)

## Tests performed

run accel_brake_map_calibrator and play a rosbag data.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by TakaHoribe** [comment]
I rebased to autowarefoundation/main. The original code was saved in my fork just for in case.
https://github.com/TakaHoribe/autoware.universe/tree/feature/improve_map_calibrator-ORIGINAL

File: vehicle/accel_brake_map_calibrator/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp [hunk 1]
```diff
@@ -791,6 +829,128 @@ void AccelBrakeMapCalibrator::executeUpdate(
                  .emplace_back(measured_acc);
 }
 
+bool AccelBrakeMapCalibrator::updateFourCellAroundOffset(
```
Comment (TakaHoribe) [inline vehicle/accel_brake_map_calibrator/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:832]: There are so many if-condition for accel_mode in this function. 
Please define variables for the accel_mode at the top of the function, and remove the if-condition from the algorithm code, like below:

before
```
  const double zll = accel_mode ? update_accel_map_value_.at(accel_pedal_index).at(accel_vel_index)
                                : update_brake_map_value_.at(brake_pedal_index).at(brake_vel_index);
  const double zhl = accel_mode
                       ? update_accel_map_value_.at(accel_pedal_index + 1).at(accel_vel_index)
                       : update_brake_map_value_.at(brake_pedal_index + 1).at(brake_vel_index);
  const double zlh = accel_mode
                       ? update_accel_map_value_.at(accel_pedal_index).at(accel_vel_index + 1)
                       : update_brake_map_value_.at(brake_pedal_index).at(brake_vel_index + 1);
  const double zhh = accel_mode
                       ? update_accel_map_value_.at(accel_pedal_index + 1).at(accel_vel_index + 1)
                       : update_brake_map_value_.at(brake_pedal_index + 1).at(brake_vel_index + 1);
```

after
```
  auto & update_map_value = accel_mode ? update_accel_map_value_ : update_brake_map_value_;
  const auto & pedal_index = accel_mode ? accel_pedal_index : brake_pedal_index;
  const auto & vel_index = accel_mode ? accel_vel_index : brake_vel_index;

  const double zll = update_map_value.at(pedal_index).at(vel_index);
  const double zhl = update_map_value.at(pedal_index + 1).at(vel_index);
  const double zlh = update_map_value.at(pedal_index).at(vel_index + 1);
  const double zhh = update_map_value.at(pedal_index + 1).at(vel_index + 1);
```

**Comment by shmpwk** [comment]
spell miss 
https://github.com/autowarefoundation/autoware.universe/actions/runs/3239609749/jobs/5309193223

I fixed it by
https://github.com/autowarefoundation/autoware.universe/pull/1255/commits/a39884bef566f56b863ed79c2304414606da2e19

**Comment by shmpwk** [comment]
LGTM!

view_plot
[accel_map.csv](https://github.com/autowarefoundation/autoware.universe/files/9897250/accel_map.csv)
[brake_map.csv](https://github.com/autowarefoundation/autoware.universe/files/9897251/brake_map.csv)

**Review by shmpwk — APPROVED** [review]

**Review by TakaHoribe — APPROVED** [review]

**Review by tkimura4 — APPROVED** [review]
LGTM

**Pull request merged**